qiskit/utils/deprecate_pulse.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Deprecation functions for Qiskit Pulse. To be removed in Qiskit 2.0.
+"""
+
+import warnings
+import functools
+
+from qiskit.utils.deprecation import deprecate_func, deprecate_arg
+
+
+def deprecate_pulse_func(func):
+    """Deprecation message for functions and classes"""
+    return deprecate_func(
+        since="1.3",
+        package_name="Qiskit",
+        removal_timeline="in Qiskit 2.0",
+        additional_msg="The entire Qiskit Pulse package is being deprecated "
+        "and will be moved to the Qiskit Dynamics repository: "
+        "https://github.com/qiskit-community/qiskit-dynamics",
+    )(func)
+
+
+def deprecate_pulse_dependency(*args, **kwargs):
+    """Deprecation message for functions and classes which use or depend on Pulse"""
+    decorator = deprecate_func(
+        since="1.3",
+        package_name="Qiskit",
+        removal_timeline="in Qiskit 2.0",
+        additional_msg="The entire Qiskit Pulse package is being deprecated "
+        "and this is a dependency on the package.",
+        **kwargs,
+    )
+
+    # Taken when `deprecate_pulse_dependency` is used with no arguments and with empty parentheses,
+    # in which case the decorated function is passed
+    if args:
+        return decorator(args[0])
+    return decorator
+
+
+def deprecate_pulse_arg(arg_name, **kwargs):
+    """Deprecation message for arguments related to Pulse"""
+    return deprecate_arg(
+        name=arg_name,
+        since="1.3",
+        package_name="Qiskit",
+        removal_timeline="in Qiskit 2.0",
+        additional_msg="The entire Qiskit Pulse package is being deprecated "
+        "and this argument uses a dependency on the package.",
+        **kwargs,
```

Comment:
Same comment as above:

```suggestion
        additional_msg="The entire Qiskit Pulse package is being deprecated "
        "and this argument uses a dependency on the package. The relevant components ",
         "for low-level pulse control have been moved to the Qiskit Dynamics ",
         "repository:  https://github.com/qiskit-community/qiskit-dynamics",
        **kwargs,
```

Reply:
All deprecated args will be removed without alternatives, in the same spirit of the suggestion in [your comment above](https://github.com/Qiskit/qiskit/pull/13164#discussion_r1800975738), we should say this explicitly for the deprecated args?